packages/identity/src/identity/ed25519.ts

```diff
@@ -27,53 +28,20 @@ export class Ed25519PublicKey implements PublicKey {
   // The length of Ed25519 public keys is always 32 bytes.
   private static RAW_KEY_LENGTH = 32;
 
-  // Adding this prefix to a raw public key is sufficient to DER-encode it.
-  // See https://github.com/dfinity/agent-js/issues/42#issuecomment-716356288
-  private static DER_PREFIX = Uint8Array.from([
-    ...[48, 42], // SEQUENCE
-    ...[48, 5], // SEQUENCE
-    ...[6, 3], // OBJECT
-    ...[43, 101, 112], // Ed25519 OID
-    ...[3], // OBJECT
-    ...[Ed25519PublicKey.RAW_KEY_LENGTH + 1], // BIT STRING
-    ...[0], // 'no padding'
-  ]);
-
   private static derEncode(publicKey: BinaryBlob): DerEncodedBlob {
-    if (publicKey.byteLength !== Ed25519PublicKey.RAW_KEY_LENGTH) {
-      const bl = publicKey.byteLength;
-      throw new TypeError(
-        `ed25519 public key must be ${Ed25519PublicKey.RAW_KEY_LENGTH} bytes long (is ${bl})`,
-      );
+    // TODO: Validation before encoding is weird, is this necessary?
```

Comment:
Opinions?

Reply:
I think the check could be seen as making sense for the restricted encoding that could only handle fixed-length byte strings. I think it does not make sense anymore. I would then probably also delete the corresponding test.